tests/test_trainer.py

```diff
@@ -28,14 +29,17 @@
 
 if is_torch_available():
     import torch
-    from torch.utils.data import IterableDataset
+    from torch.utils.data import Dataset, IterableDataset
```

Comment:
No reason to import `Dataset` from torch.

Reply:
The type hinting for the arguments of `Trainer.get_eval_dataloader` indicates `Dataset`for the `eval_dataset` argument.
I changed `RegressionDataset` to `RegressionDataset(Dataset)`.

Reply:
The type hint does what it can to indicate the dataset must have a len (not even true anymore) and a getitem. There are plenty of things that implement this without subclassing `torch.utils.data.Dataset` (which should really be a `Protocol` form Python 3.8 and not a class).